### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by liucuxiu**
- add cors
- add mongo for user api

### CODE REVIEW

File: src/users/api/controllers/updateProfileInfo/UpdateProfileInfoController.ts
Comment (thetnaingtta) [inline src/users/api/controllers/updateProfileInfo/UpdateProfileInfoController.ts:1]: why do we need this console log in the controller? Is that we check in our local and remove if everything is ok?

Comment (liucuxiu) [reply src/users/api/controllers/updateProfileInfo/UpdateProfileInfoController.ts:1]: oh I forgot to remove :D thank you

File: src/app.ts [hunk 1]
```diff
@@ -13,6 +14,10 @@ mongoose.connect(mongoUri).then(() => {
   console.log('connected Mongo!');
```
Comment (sithwin) [inline src/app.ts:15]: should use better logging

File: src/authService/middleware/auth0.middleware.ts [hunk 2]
```diff
@@ -4,6 +4,9 @@ import config from 'config';
 const validateJwt = auth({
   issuerBaseURL: config.get('auth0.domain'),
   audience: config.get('auth0.audience'),
+  jwksUri: 'https://liucuxiu.auth0.com/.well-known/jwks.json'
```
Comment (sithwin) [inline src/authService/middleware/auth0.middleware.ts]: we should have our own auth0 :)

File: src/users/repos/createUser/impl/MongoCreateUserRepository.spec.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+jest.mock('../../../database/mongo/models/User.model');
+
+describe('MongoCreateUserRepository', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should create a user in the database', async () => {
+    // Arrange
```
Comment (sithwin) [inline src/users/repos/createUser/impl/MongoCreateUserRepository.spec.ts]: this comment necessary?

File: src/users/repos/getUserById/impl/MongoGetUserByIdRepository.spec.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    expect(UserModel.findById).toHaveBeenCalledWith(mockUserId);
+  });
+
+  it('should return undefined if user with ID is not found', async () => {
+    const nonExistentUserId = 'nonExistentUserId';
+
+    findByIdMock.mockResolvedValue(null);
+
+    await expect(getUserByIdRepository.getUserById(nonExistentUserId)).rejects.toThrow('User not found');
+    expect(UserModel.findById).toHaveBeenCalledWith(nonExistentUserId);
```
Comment (sithwin) [inline src/users/repos/getUserById/impl/MongoGetUserByIdRepository.spec.ts:58]: should follow consistence line break

File: src/users/repos/getUserById/impl/MongoGetUserByIdRepository.ts [hunk 5]
```diff
@@ -3,13 +3,16 @@ import { UserModel } from '../../../database/mongo/models/User.model';
...
 export class MongoGetUserByIdRepository implements IGetUserByIdRepository {
-  async getUserById(id: string): Promise<User | undefined> {
+  async getUserById(id: string): Promise<User> {
     const result = await UserModel.findById(id);
-    if (!result) return undefined;
 
-    const user = User.create(
+    if (!result) throw new Error('User not found');
+
+    // need to enhance this
```
Comment (sithwin) [inline src/users/repos/getUserById/impl/MongoGetUserByIdRepository.ts]: should comment as TODO

File: src/users/repos/updateUser/impl/MongoUpdateUserByIdRepository.spec.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      imageUrl: 'https://example.com/updated-image.jpg',
+      address: '456 New St',
+      address_2: 'Apt 789',
+      phone: '987-654-3210',
+    };
+
+    findByIdAndUpdateMock.mockResolvedValue(null);
+
+    await expect(updateUserRepository.updateUser(mockUserId, mockUpdateData)).rejects.toThrow('User not found');
+    expect(UserModel.findByIdAndUpdate).toHaveBeenCalledWith(mockUserId, mockUpdateData);
```
Comment (sithwin) [inline src/users/repos/updateUser/impl/MongoUpdateUserByIdRepository.spec.ts:52]: should follow consistence line break

**Comment by liucuxiu** [comment]
I have updated the PR, pls review, thanks

**Review by thetnaingtta — APPROVED** [review]
Good job!

**Pull request merged**